### PR TITLE
Add avx2_vnni_2 check for bf16 and fp16 support on SRF

### PR DESCRIPTION
**Pull request by CaoE**
* add avx2_vnni_2 check for bf16 and fp16 support on SRF
* add bf16/f16 backward check to explicitly throw message that bf16/f16 backward is not supported on SRF. Add a new check `IDEEP_CHECK` instead of enabling `IDEEP_ENFORCE` in release mode. It is because enabling `IDEEP_ENFORCE` will reduce the performance of RN50 real-time inference by ~4%.

### CODE REVIEW

**Comment by CaoE** [comment]
Related tests passed: https://github.com/pytorch/pytorch/pull/108634

**Comment by CaoE** [comment]
@jgong5 @XiaobingSuper @Xia-Weiwen Could you please review this PR ? https://github.com/pytorch/pytorch/pull/108634 have passed PyTorch CI with this PR.

File: include/ideep/abstract_types.hpp [hunk 1]
```diff
@@ -70,13 +83,20 @@ static bool has_bf16_type_support() {
...
-      dnnl::get_effective_cpu_isa() >= dnnl::cpu_isa::avx512_core;
+      dnnl::get_effective_cpu_isa() >= dnnl::cpu_isa::avx2_vnni_2;
   return support_bf16;
 }
 
+static bool check_isa_avx2_vnni_2() {
+  static bool is_avx2_vnni_2 =
+      dnnl::get_effective_cpu_isa() == dnnl::cpu_isa::avx2_vnni_2;
+  return is_avx2_vnni_2;
+}
```
Comment (jgong5) [inline include/ideep/abstract_types.hpp]: What's the intention of this API? Checking if the CPU supports avx2 vnni 2 and not others? If so, should we rename the function as `check_isa_is_avx2_vnni_2`?

File: include/ideep/operators/batchnorm.hpp [hunk 2]
```diff
@@ -192,6 +192,10 @@ struct batch_normalization_backward
           batch_normalization_flag::use_scale |
           batch_normalization_flag::use_shift,
       const engine& aengine = engine::cpu_engine()) {
+    IDEEP_CHECK(!(check_isa_avx2_vnni_2() &&
+                  utils::one_of(src.get_data_type(),
+                                data_type::bf16, data_type::f16)),
+                  "DNNL does not support bf16/f16 backward on the platform with avx2_vnni_2");
```
Comment (jgong5) [inline include/ideep/operators/batchnorm.hpp]: Is it temporary due to onednn implementation?

File: include/ideep/abstract_types.hpp (hunk 1 above)
Comment (CaoE) [reply include/ideep/abstract_types.hpp]: Fixed

File: include/ideep/operators/batchnorm.hpp (hunk 2 above)
Comment (CaoE) [reply include/ideep/operators/batchnorm.hpp]: It is not temporary since oneDNN will not support bf16/fp16 backward on SRF.

**Review by jgong5 — APPROVED** [review]
